background/web3.js

```diff
@@ -60,20 +62,38 @@ export default class Web3 {
   web3Message (msg) {
     const { payload, origin } = msg;
 
-    return fetch(
-      `http://${this.DAPPS}/rpc/`,
-      {
-        method: 'POST',
-        mode: 'cors',
-        headers: new Headers({
-          'Content-Type': 'application/json',
-          'X-Parity-Origin': origin
-        }),
-        body: JSON.stringify(payload),
-        redirect: 'error',
-        referrerPolicy: 'origin'
-      }
-    ).then((response) => response.json());
+    const request = {
+      method: 'POST',
+      mode: 'cors',
+      headers: new Headers({
+        'Content-Type': 'application/json',
+        'X-Parity-Origin': origin
+      }),
+      body: JSON.stringify(payload),
+      redirect: 'error',
+      referrerPolicy: 'origin'
+    };
+    return fetch(`http://${this.DAPPS}/rpc/`, request)
+      .catch(err => {
+        // TODO [ToDr] Get rid of this in the future!
+        // Version 1.7 of Parity runs dapps server on the same port as RPC.
+        // Previous versions were running on :8080.
+        // For backward compatibility we support both cases.
+        const defaultPort = ':' + DEFAULT_CONFIG.DAPPS.split(':')[1];
+        const newParityDappsPort = ':8545';
+        if (!(err.message === FAILURE && this.DAPPS.endsWith(defaultPort))) {
+          throw err;
+        }
+
+        const newDapps = this.DAPPS.replace(defaultPort, newParityDappsPort);
+        const url = `http://${newDapps}/rpc/`;
+        return fetch(url, request).then(res => {
+          // Update dapps if it succeeds
+          this.DAPPS = newDapps;
```

Comment:
I don't really want to overwrite user settings now. Switching between different Parity versions would be a pain in the ass (cause switching to new one would overwrite the config and either old one stops working or we need to do reverse check as well).

We can override config settings when Parity 1.7 get's more stable.